ui/src/app/workflows/components/resubmit-workflow-panel.tsx

```diff
@@ -12,103 +12,93 @@ interface Props {
     isArchived: boolean;
 }
 
-interface State {
-    overrideParameters: boolean;
-    workflowParameters: Parameter[];
-    memoized: boolean;
-    error?: Error;
-    isSubmitting: boolean;
-}
+export function ResubmitWorkflowPanel(props: Props) {
+    const [overrideParameters, setOverrideParameters] = useState(false);
+    const [workflowParameters, setWorkflowParameters] = useState<Parameter[]>(JSON.parse(JSON.stringify(props.workflow.spec.arguments.parameters || [])));
+    const [memoized, setMemoized] = useState(false);
+    const [error, setError] = useState<Error>();
+    const [isSubmitting, setIsSubmitting] = useState(false);
 
-export class ResubmitWorkflowPanel extends React.Component<Props, State> {
-    constructor(props: any) {
-        super(props);
-        const state: State = {
-            workflowParameters: JSON.parse(JSON.stringify(this.props.workflow.spec.arguments.parameters || [])),
-            memoized: false,
-            isSubmitting: false,
-            overrideParameters: false
+    const submit = () => {
```

Comment:
```suggestion
    function submit() {
```

prefer named function over const assigned to anonymous function (it's more performant and easier to trace in debug logs etc since "anonymous" is not the name -- I also used this in all of my recent PRs)